tests/wpunit/AccessFunctionsTest.php

```diff
@@ -701,4 +701,380 @@ public function testRegisterMutationWithIsPrivatePreventsPublicMutationExecution
+			'PostFormatToPostConnectionWhereArgs',
+			'PostFormatToPostConnection',
+			'PostFormatToPostConnectionEdge',
+			'PostToPreviewConnectionEdge',
+			'PostToRevisionConnectionWhereArgs',
+			'PostToRevisionConnection',
+			'PostToRevisionConnectionEdge',
+			'PostToTagConnectionWhereArgs',
+			'PostToTagConnection',
+			'PostToTagConnectionEdge',
+			'TagToPostConnectionWhereArgs',
+			'TagToPostConnection',
+			'TagToPostConnectionEdge',
+			'PostToTermNodeConnectionWhereArgs',
+			'PostToTermNodeConnection',
+			'PostToTermNodeConnectionEdge',
+			'CategoryToPostConnectionWhereArgs',
+			'CategoryToPostConnection',
+			'CategoryToPostConnectionEdge',
+			'PostIdType',
+			'RootQueryToPostConnectionWhereArgs',
+			'RootQueryToPostConnection',
+			'RootQueryToPostConnectionEdge',
+		];
+
+		$this->assertEmpty( array_intersect( $types, $removed_types ) );
+
+		// Ensure connection is removed.
+		$query = '
+			{
+				categories {
+					nodes {
+						posts {
+							nodes {
+								__typename
+							}
+						}
+					}
+				}
+			}
+		';
+
+		$actual = $this->graphql( compact( 'query') );
+		
+		$this->assertIsValidQueryResponse( $actual );
+		$this->assertArrayHasKey( 'errors', $actual );
+
+		// Ensure Post is removed from interfaces.
+		$query = '
+		{
+			__type(name: "ContentNode") {
+				possibleTypes {
+					name
+				}
+			}
+		}
+		';
+		$actual = $this->graphql( compact( 'query' ) );
+
+		$this->assertIsValidQueryResponse( $actual );
+		$this->assertArrayNotHasKey( 'errors', $actual );
+		$this->assertNotContains( 'Post', array_column( $actual['data']['__type']['possibleTypes'], 'name' ) );
+
+		// Ensure Post is removed from unions.
+		$query = '
+		{
+			__type(name: "MenuItemObjectUnion") {
+				possibleTypes {
+					name
+				}
+			}
+		}
+		';
+
+		$actual = $this->graphql( compact( 'query' ) );
+
+		$this->assertIsValidQueryResponse( $actual );
+		$this->assertArrayNotHasKey( 'errors', $actual );
+		$this->assertNotContains( 'Post', array_column( $actual['data']['__type']['possibleTypes'], 'name' ) );
+
+		/**
+		 * Since only the Post Type is removed, but the connection is still registered, we should expect the resolvers to still return an unresolved type.
+		 */
+		$post_id = $this->factory()->post->create(
+			[
+				'post_title' => 'Test deregister type',
+				'post_status' => 'publish',
+			]
+		);
+
+		$query = '
+		{
+			contentNodes {
+				nodes {
+					id
+				}
+			}
+			contentTypes {
+				nodes {
+					name
+				}
+			}
+		}
+		';
+
+		$actual = $this->graphql( compact( 'query') );
+
+		$this->assertIsValidQueryResponse( $actual );
+		$this->assertArrayHasKey( 'errors', $actual );
+		$this->assertStringContainsString( 'GraphQL Interface Type `ContentNode` returned `null', $actual['errors'][0]['debugMessage'] );
+		$this->assertNull( $actual['data']['contentNodes']['nodes'][0] );
+		$this->assertContains( 'post', array_column( $actual['data']['contentTypes']['nodes'], 'name' ) );
```

Comment:
This is an important caveat to the functionality. See note 4in the PR description comments.